### PR TITLE
implementing baseline code for socker.io

### PR DESCRIPTION
**Pull request by mdmontes**

### CODE REVIEW

**Review by Abdulaiman — APPROVED** [review]
Great work! The changes in this pull request look good to me. I've reviewed the code and it seems to be in line with our project's standards.

File: src/server.ts [hunk 1]
```diff
@@ -1,6 +1,11 @@
 const mongoose = require("mongoose");
 const dotenv = require("dotenv");
 const app = require("./app.ts");
+const socket = require("socket.io");
+import { Socket } from "socket.io";
```
Comment (royemosby) [inline src/server.ts:5]: What are these two different types of import statements doing?

**Review by royemosby — CHANGES_REQUESTED** [review]
Requesting minor change and had one question about imports

File: src/server.ts (hunk 1 above)
Comment (mdmontes) [reply src/server.ts:5]: Hey Roy,

The Lower case "socket" is the socket object that we are using to emit and listen to messages throughout the code. The upper case "Socket" is a type that had to import and reference due to this file being in TypeScript in order to specify the "type" of object corresponding to lower case "socket". They are being used at the very beginning of this script:

`io.on("connection", (socket: Socket) => {
  console.log('we should have a connection');

  socket.on("addUser", (userId) => {
    onlineUsers.set(userId, socket.id);
    socket.emit("getUsers", Array.from(onlineUsers));
  });
  
  socket.on("sendMessage", (data) => {
    const sendUserSocket = onlineUsers.get(data.to);
    if (sendUserSocket) {
      socket.to(sendUserSocket).emit("getMessage", data);
    }
  });

  socket.on("disconnect", () => {
    // Remove the disconnected socket from onlineUsers map
    for (const [userId, socketId] of onlineUsers) {
      console.log('this is getting activated')
      if (socketId === socket.id) {
        console.log(`deleting data from socket ${socketId}`)
        onlineUsers.delete(userId);
        break;
      }
    }
    console.log(`Socket disconnected: ${socket.id}`);
    console.log(`Updated onlineUsers: ${JSON.stringify(Array.from(onlineUsers))}`);
  })
});`

**Review by jaheemprevost — APPROVED** [review]

**Review by royemosby — APPROVED** [review]
GTG!

**Pull request merged**